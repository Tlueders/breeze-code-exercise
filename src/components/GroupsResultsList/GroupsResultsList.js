import React, { Component } from 'react'
import { Table } from 'semantic-ui-react'
import { groupBy } from '../../../../../Library/Caches/typescript/3.6/node_modules/rxjs/internal/operators/groupBy';

class PeopleResultsList extends Component {
    constructor(props) {
        super(props);
        this.state = { data: [] };
    }

    componentDidMount() {
        fetch("http://localhost:8000/api/groups")
          .then(response => response.json())
          .then(data => this.setState({ data: data.data }));
    }

    render() {
        var data = this.state.data || [];

        return (
            <Table celled padded>
              <Table.Header>
                <Table.Row>
                  <Table.HeaderCell singleLine>Group Name</Table.HeaderCell>
                  <Table.HeaderCell>Group Members</Table.HeaderCell>
                </Table.Row>
              </Table.Header>

              <Table.Body>

              {
                  data.map((group, index) => {
                      return (
                          <Table.Row key={index}>
                              <Table.Cell singleLine>{ group.group_name }</Table.Cell>
                              <Table.Cell singleLine>{ group.members }</Table.Cell>
                          </Table.Row>
                      );
                    })
              }

              </Table.Body>
            </Table>
    );
}

}

export default PeopleResultsList;
