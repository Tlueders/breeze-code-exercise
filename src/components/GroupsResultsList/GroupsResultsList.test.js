import React from 'react';
import { shallow } from 'enzyme';
import GroupsResultsList from './GroupsResultsList';

let wrapper, data;

describe('<GroupsResultsList />', () => {

    beforeAll(() => {
        wrapper = shallow(<GroupsResultsList />)
        data = [{
            "id": 1,
            "group_name": "Volunteers",
            "created_at": "2019-10-08 18:01:01",
            "updated_at": "2019-10-08 18:01:01",
            "person": [
                {
                    "id": 132,
                    "first_name": "Macie",
                    "last_name": "Emmerich",
                    "email_address": "cremin.marjory@hotmail.com",
                    "status": "active",
                    "group_id": 1,
                    "updated_at": "2019-07-20 22:05:47",
                    "created_at": "2019-07-20 22:05:47"
                }
            ]
        }]

        wrapper.setState({ 'data' : data });
    });

    test('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
