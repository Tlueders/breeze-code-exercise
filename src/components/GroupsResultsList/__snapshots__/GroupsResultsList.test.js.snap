// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupsResultsList /> should match the snapshot 1`] = `
<Table
  as="table"
  celled={true}
  fixed={true}
  padded={true}
  sortable={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
        onClick={[Function]}
        sorted={null}
      >
        Group ID
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        onClick={[Function]}
        sorted={null}
      >
        Group Name
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Group Members
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
      key="0"
    >
      <TableCell
        as="td"
      >
        1
      </TableCell>
      <TableCell
        as="td"
      >
        Volunteers
      </TableCell>
      <TableCell
        as="td"
      >
        <ul>
          <li
            key="0"
          >
            Macie Emmerich
          </li>
        </ul>
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
